Source/Request.swift

```diff
@@ -557,10 +557,8 @@ public class Request {
     func reset() {
         error = nil
 
-        uploadProgress.totalUnitCount = 0
-        uploadProgress.completedUnitCount = 0
-        downloadProgress.totalUnitCount = 0
-        downloadProgress.completedUnitCount = 0
+        uploadProgress = Progress(totalUnitCount: 0)
```

Comment:
Great if this fixes the crash, but it makes `Progress` useless, no?

Reply:
This is in the `reset()` function, so I’m not sure why the total and completed unit count would both need to be set to 0 instead of the progress object just recreated.

In any case yes, this appears to fix the crash.

Reply:
Don't you have to use a single `Progress` instance for it to work? If I'm observing the `Progress` instance, or composing it in other `Progress` values, doesn't it need to stay the same instance to work? I've never actually used this API, but I can't see any other way for it to work.

Reply:
In any case, this is another nice bug to report.

Reply:
Yep, I’ll try to build a small reproduction.

Reply:
Filed as [SR-12391](https://bugs.swift.org/browse/SR-12391).